bitnami/flux/templates/source-controller/deployment.yaml

```diff
@@ -0,0 +1,215 @@
+      {{- else }}
+      affinity:
+        podAffinity: {{- include "common.affinities.pods" (dict "type" .Values.sourceController.podAffinityPreset "component" "source-controller" "context" $) | nindent 10 }}
+        podAntiAffinity: {{- include "common.affinities.pods" (dict "type" .Values.sourceController.podAntiAffinityPreset "component" "source-controller" "context" $) | nindent 10 }}
+        nodeAffinity: {{- include "common.affinities.nodes" (dict "type" .Values.sourceController.nodeAffinityPreset.type "key" .Values.sourceController.nodeAffinityPreset.key "values" .Values.sourceController.nodeAffinityPreset.values) | nindent 10 }}
+      {{- end }}
+      {{- if .Values.sourceController.nodeSelector }}
+      nodeSelector: {{- include "common.tplvalues.render" ( dict "value" .Values.sourceController.nodeSelector "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.sourceController.tolerations }}
+      tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.sourceController.tolerations "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.sourceController.priorityClassName }}
+      priorityClassName: {{ .Values.sourceController.priorityClassName | quote }}
+      {{- end }}
+      {{- if .Values.sourceController.schedulerName }}
+      schedulerName: {{ .Values.sourceController.schedulerName | quote }}
+      {{- end }}
+      {{- if .Values.sourceController.topologySpreadConstraints }}
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.sourceController.topologySpreadConstraints "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.sourceController.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.sourceController.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.sourceController.terminationGracePeriodSeconds }}
+      terminationGracePeriodSeconds: {{ .Values.sourceController.terminationGracePeriodSeconds }}
+      {{- end }}
+      {{- if or (and .Values.volumePermissions.enabled .Values.sourceController.persistence.enabled) .Values.sourceController.initContainers }}
+      initContainers:
+        {{- if and .Values.volumePermissions.enabled .Values.sourceController.persistence.enabled }}
+        - name: volume-permissions
+          image: {{ include "flux.volumePermissions.image" . }}
+          imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
+          command:
+            - /bin/bash
+          args:
+            - -ec
+            - |
+              mkdir -p {{ .Values.sourceController.persistence.mountPath }}
+              find {{ .Values.sourceController.persistence.mountPath }} -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | xargs -r chown -R {{ .Values.sourceController.containerSecurityContext.runAsUser }}:{{ .Values.sourceController.podSecurityContext.fsGroup }}
```

Comment:
I am not sure about the content of the directory `.Values.sourceController.persistence.mountPath` but, does this need to be executed in each restart of the pod ?